akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -44,17 +45,27 @@ trait Projection[StreamElement] {
    * @return A [[scala.concurrent.Future]] that represents the asynchronous completion of the user EventHandler
    *         function.
    */
-  def processElement(elt: StreamElement): Future[StreamElement]
+  def processElement(elt: StreamElement): Future[Done]
```

Comment:
I would like to reassess the need for returning `Future[StreamElement]` here. 

If we want it, it means that we must take the user function `Elt => Future[Done]`, map back to  `Elt => Future[StreamElement]` because of this method and then once again to `Elt => Future[Done]` because of `mappedSource`.

If we don't have a stronger reason for that, I wouid prefer to keep it on `Future[Done]`

Reply:
@ignasi35, can you comment on this?

Reply:
If `StreamElement` is a type parameter (and not a well-known type on the `akka-projection` API) using `Future[StreamElement]` is not necessary because the bytecode instrumentation doesn't know the actual type beforehand.

Reply:
Ok, so it doesn't matter. Let's keep `Future[Done]`